docs/guides/validation-user-guide.md

```diff
@@ -4,4 +4,461 @@ headline: Validation User Guide
+and `UserIdentity.twitter` must be set.
+
+<p class="warning">
+`(is_required)` option is not yet supported in Dart.
+</p>
+
+In some other cases, a field may be either required or not, depending on the value of another field.
+Consider an example of an online store item:
+
+```proto
+import "spine/options.proto";
+
+// A product which can be purchased at the online store.
+message Item {
+
+    // ...
+
+    Timestamp when_opened_for_sale = 42;
+    UserId who_opened_for_sale = 43 [(goes).with = "when_opened_for_sale"];
+}
+```
+
+The `Item.who_opened_for_sale` field only makes sense for the domain if 
+the `Item.when_opened_for_sale` field is set. If `who_opened_for_sale` is set and
+`when_opened_for_sale` is not, a constraint violation is produced.
+
+Finally, there are some cases, in which a pair of fields may be set at the same time, but at least
+one of them must be set. This and more complex cases are handled by the type-level
+`(required_field)` option:
+
+```proto
+import "spine/options.proto";
+
+// A name of a person.
+message PersonName {
+    option (required_field) = "given_name|honorific_prefix & family_name";
+
+    string honorific_prefix = 1;
+    string given_name = 2;
+    string middle_name = 3;
+    string family_name = 4;
+    string honorific_suffix = 5;
+}
+``` 
+
+In case of `PersonName`, either `given_name` or both `honorific_prefix` and `family_name` must be
+set. All three can be set at the same time.
+
+#### Missing fields
+
+In case if a required field is missing, the validation error message will explicitly say so.
+However, if you need a specific error message for this field, you can provide it via
+the `(if_missing)` option:
+
+```proto
+import "spine/options.proto";
+
+// A phone number represented by a string of digits.
+message PhoneNumber {
+    string digits = 1 [(required) = true,
+                       (if_missing).msg_format = "Phone number must contain digits."];
+}
+```
+
+Note that this option only applies to fields marked with `(required)` and not to the fields
+referenced via any other options.
+
+If `(goes)` option is used, the error message can be customized with the `(goes).msg_format`
+parameter. Note that the message should contain two "`%s`" insertion points: first for the name of
+the field declaring the option and second for the name of the field targeted by the option.
+
+#### When `(required)` is implicit
+
+When defining the domain [Commands](../introduction/naming-conventions.html#command-definitions),
+[Events](../introduction/naming-conventions.html#event-definitions) or entity states, we have found
+to be convenient that the first field of the respective Message is the identifier. Therefore, by
+convention, Spine treats the first fields of such objects as their IDs:
+
+```proto
+import "spine/options.proto";
+
+// The state of the User Aggregate.
+message User {
+    option (entity).kind = AGGREGATE;
+
+    UserId id = 2;
```

Comment:
The problem with this sample is that we need more imports, and package for `UserId` type. Otherwise, it's going to be misleading for the reader.